test/unit/api/term_black.cpp

```diff
@@ -684,6 +684,13 @@ TEST_F(TestApiTermBlack, getInteger)
   Term int9 = d_solver.mkInteger("4294967296");
   Term int10 = d_solver.mkInteger("18446744073709551615");
   Term int11 = d_solver.mkInteger("18446744073709551616");
+  Term int12 = d_solver.mkInteger("-0");
+
+  ASSERT_THROW(d_solver.mkInteger(""), CVC4ApiException);
+  ASSERT_THROW(d_solver.mkInteger("-"), CVC4ApiException);
+  ASSERT_THROW(d_solver.mkInteger("-1-"), CVC4ApiException);
+  ASSERT_THROW(d_solver.mkInteger("0.0"), CVC4ApiException);
+  ASSERT_THROW(d_solver.mkInteger("-0.1"), CVC4ApiException);
```

Comment:
What should happen with "-0" or leading zeroes like "012", "0000"?

Reply:
Not sure the current API accepts leading zeros without problems. The smt document says this for numerals
```
Numerals. A〈numeral〉is the digit 0 or a non-empty sequence of digits not starting with 0.
```

Reply:
OK then we should probably be more strict and not allow "01", "00", "-0",...

Reply:
We can do that and also strictly follow the SMTLIB standard in the API
For example `(-1)` is not allowed, whereas `(- 1)` is allowed. Should we also apply this to the the API as well?

```java
Term one = d_solver.mkInteger("1");
Term negativeOne = d_solver.mkTerm(UMINUS, one);
```

```java
Term negativeOne = d_solver.mkInteger("-1")
```

Reply:
No that should both be possible, that's a usability question of `mkInteger`. The API can allow more than what SMT-LIB allows.

Reply:
Ok. After this PR leading zeros would be unsupported.